api/src/operations/read/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Accountability, PrimaryKey } from '@directus/shared/types';
+import { defineOperationApi, toArray } from '@directus/shared/utils';
+import { ItemsService } from '../../services';
+import { Item } from '../../types';
+import { getAccountabilityForRole } from '../../utils/get-accountability-for-role';
+import { parseJSON } from '../../utils/parse-json';
+
+type Options = {
+	mode: 'one' | 'many' | 'query';
```

Comment:
This was the next best thing to implement a basic `read` operation. But, as the "mode" could also be determined by the type of the key (`readOne` if key is a single key, `readMany` if key is an array of keys, `readByQuery` if key is `null`), this option feels a little bit redundant. Thoughts on removing the option?

Reply:
This sounds great! And possibly addresses @erondpowell's question regarding the multiple keys even for `one` scenario. But I do wonder how would we be able to toggle the query input in this case? (I believe aside from read, currently cases like `delete` would only show the query input when the mode is `query`)

Reply:
Good question! I have to admit that I haven't thought about those cases. For read it's obvious because `readOne` and `readMany` allow you to specify a query in addition to the key/keys.
We could always show the query input and just ignore it if the key option has been set. But that might actually be more confusing than having a `mode` option.

Reply:
Yeah this is more so a UX question 🤔 maybe somehow subdue the other input to signal that it is being ignored. Not super sure as well atm.